### PR TITLE
schema and feature compilers for project-specific preferences

### PR DESCRIPTION
**Pull request by gireeshpunathil**
the project specific preferences property page contained only plugin compiler (manifest compiler), while the schema and feature compiler settings are available in the workspace-wide settings. remove this restriction and open up all the compiler settings for project-specific settings too.

a screenshot of thew new project-specific settings:

image

Fixes: https://github.com/eclipse-pde/eclipse.pde/issues/464

### CODE REVIEW

**Comment by gireeshpunathil** [comment]
I haven't learned interpreting the test results yet; I see the below failure. not sure if this is related to this PR:

```
org.eclipse.pde.ui.tests.launcher.FeatureBasedLaunchTest  Time elapsed: 1.33 s  <<< FAILURE!
java.lang.AssertionError: An internal error occurred during: "Load Target Platform"
at org.eclipse.pde.ui.tests.util.TargetPlatformUtil.loadAndSetTarget(TargetPlatformUtil.java:112)
at org.eclipse.pde.ui.tests.util.TargetPlatformUtil.createAndSetTarget(TargetPlatformUtil.java:148)
at org.eclipse.pde.ui.tests.util.TargetPlatformUtil.setRunningPlatformSubSetAsTarget(TargetPlatformUtil.java:91)
at org.eclipse.pde.ui.tests.util.TargetPlatformUtil.setRunningPlatformAsTarget(TargetPlatformUtil.java:80)
at org.eclipse.pde.ui.tests.launcher.AbstractLaunchTest.setupTargetPlatform(AbstractLaunchTest.java:62)
```

**Comment by vik-chand** [comment]
> I haven't learned interpreting the test results yet; I see the below failure. not sure if this is related to this PR:
> 
> ```
> org.eclipse.pde.ui.tests.launcher.FeatureBasedLaunchTest  Time elapsed: 1.33 s  <<< FAILURE!
> java.lang.AssertionError: An internal error occurred during: "Load Target Platform"
> at org.eclipse.pde.ui.tests.util.TargetPlatformUtil.loadAndSetTarget(TargetPlatformUtil.java:112)
> at org.eclipse.pde.ui.tests.util.TargetPlatformUtil.createAndSetTarget(TargetPlatformUtil.java:148)
> at org.eclipse.pde.ui.tests.util.TargetPlatformUtil.setRunningPlatformSubSetAsTarget(TargetPlatformUtil.java:91)
> at org.eclipse.pde.ui.tests.util.TargetPlatformUtil.setRunningPlatformAsTarget(TargetPlatformUtil.java:80)
> at org.eclipse.pde.ui.tests.launcher.AbstractLaunchTest.setupTargetPlatform(AbstractLaunchTest.java:62)
> ```

This looks unlikely from the PR. Rebuild the pr check.

**Comment by vik-chand** [comment]
> I haven't learned interpreting the test results yet; I see the below failure. not sure if this is related to this PR:
> 
> ```
> org.eclipse.pde.ui.tests.launcher.FeatureBasedLaunchTest  Time elapsed: 1.33 s  <<< FAILURE!
> java.lang.AssertionError: An internal error occurred during: "Load Target Platform"
> at org.eclipse.pde.ui.tests.util.TargetPlatformUtil.loadAndSetTarget(TargetPlatformUtil.java:112)
> at org.eclipse.pde.ui.tests.util.TargetPlatformUtil.createAndSetTarget(TargetPlatformUtil.java:148)
> at org.eclipse.pde.ui.tests.util.TargetPlatformUtil.setRunningPlatformSubSetAsTarget(TargetPlatformUtil.java:91)
> at org.eclipse.pde.ui.tests.util.TargetPlatformUtil.setRunningPlatformAsTarget(TargetPlatformUtil.java:80)
> at org.eclipse.pde.ui.tests.launcher.AbstractLaunchTest.setupTargetPlatform(AbstractLaunchTest.java:62)
> ```
If this  fails consistently, you could run this test locally. Run all the ~500 pde.ui tests on your local machine

**Comment by merks** [comment]
Other PRs also show these failures:

https://github.com/eclipse-pde/eclipse.pde/pull/518

But that PR has a green build earlier...

**Comment by gireeshpunathil** [comment]
@vik-chand - thanks.
 - looks like I don't have permission to re-run the CI. Could you pls trigger a re-run?
 - running it locally: facing some issues in doing this, but continuing to debug.

**Comment by laeubi** [comment]
> looks like I don't have permission to re-run the CI. Could you pls trigger a re-run?

The easiest way is rebase your branch on current master and (force) push it.

**Comment by vik-chand** [comment]
>>acing some issues in doing this

What is the issue  you are facing?

**Comment by gireeshpunathil** [comment]
> What is the issue you are facing?

``` Assertions cannot be resolved	AbstractLaunchTest.java	/org.eclipse.pde.ui.tests/src/org/eclipse/pde/ui/tests/launcher	line 122	Java Problem
Dimension cannot be resolved to a variable	InitializeModelsPerfTest.java	/org.eclipse.pde.ui.tests/src/org/eclipse/pde/ui/tests/performance/parts	line 40	Java Problem
No available bundle exports package 'org.assertj.core.api'	MANIFEST.MF	/org.eclipse.pde.ui.tests/META-INF	line 45	Plug-in Problem
PerformanceTestCase cannot be resolved to a type	AbstractSchemaPerfTest.java	/org.eclipse.pde.ui.tests/src/org/eclipse/pde/ui/tests/performance/parts	line 30	Java Problem
StandardRepresentation cannot be resolved to a type	AbstractLaunchTest.java	/org.eclipse.pde.ui.tests/src/org/eclipse/pde/ui/tests/launcher	line 122	Java Problem
The hierarchy of the type SchemaLoaderPerfTest is inconsistent	SchemaLoaderPerfTest.java	/org.eclipse.pde.ui.tests/src/org/eclipse/pde/ui/tests/performance/parts	line 24	Java Problem
parts	line 26	Java Problem
```

@vik-chand - these are some of the issues. I guess I am missing some bundles, but not sure what, and where are they available. any pointers?

**Comment by vik-chand** [comment]
Get everything from pde repo
Get everything from jdt.core
Get everything from jdt.binary
Get [eclipse-test-framework](https://download.eclipse.org/eclipse/downloads/drops4/I20230322-1800/download.php?dropFile=eclipse-test-framework-I20230322-1800.zip) ( edit and add this in your target)

Post this you should be able to run the tests locally

**Comment by vik-chand** [comment]
You can go to target preference page-> Select target-> Click Edit and then add the location of test framework

**Comment by vik-chand** [comment]
Looks like adding org.eclipse.test.performance to your target should fix it. It is present in eclipse-test-framework.zip

**Comment by laeubi** [comment]
@gireeshpunathil @vik-chand @merks can this be finished as of today so it goes into M1?

**Comment by gireeshpunathil** [comment]
@laeubi - thanks.

I tested this locally, and did not find any failures. Looking at recent PRs, I see many or most of those have unrelated failures in Mac and windows, so. I think this is the same. Unless there is any specific review comments, I deem this as ready to merge.

**Comment by laeubi** [comment]
@vik-chand do you like to merge this if you are fine with it even though we missed the freeze day by one ...

**Comment by laeubi** [comment]
I tried to rebase this but with no luck @gireeshpunathil can you probabbly rebase your changes on top of master? It seem to contain a lot of recent changes now so probabbly this is based on an outdated branch?

**Comment by gireeshpunathil** [comment]
tldr; it is rebased now, thanks.

@laeubi - done. something weird happened - by which I was unable to rebase in the normal manner. The `git push` did not do anything, it kept saying everything is upto-date. I had to then delete the origin branch (which caused this PR to auto-close) and then re-push.

**Comment by HannesWell** [comment]
> I'll recommend this in early M2 as it is a feature ( and big change in itself)
> 
> @HannesWell What do you think?

Agree. Furthermore I assume it is not urgent, is it?

**Review by vik-chand — APPROVED** [review]
Works well. Code looks good !
